epub33/rs/index.html

```diff
@@ -1822,33 +1837,72 @@ <h4>Scripting Considerations</h4>
-							notify the user that network activity is occurring and/or that allow them to disable it.</p>
+						<p>
+							Reading Systems that allow users to add/remove publications from a managed library 
+							(a.k.a “bookshelf”) may maintain the publication’s unique origin when the publication is
+							removed and subsequently re-imported into the content library. Conversely, Reading
+							Systems may create a new unique origin for every newly-added publication.
+						</p>
 					</li>
+
 					<li>
-						<p>If a Reading System allows users to store persistent data, it must treat that data as
+						<p>
+							The unique origin for a given publication (and by extension, for all its content documents)
+							must be preserved across "renders", i.e. when closing/re-opening publications, or when navigating back-and-forth between documents within an opened publication.
+						</p>
+					</li>
+
+					<li>
+						<p>
+							Reading Systems must ensure that the data tied to specific <a href="#confreq-rs-scripted-origin">origins</a>
+							(e.g., Web Storage [[?WEBSTORAGE]] or IndexedDB [[?INDEXEDDB]]) is preserved between "reading".
+							Reading System implementations that integrate the capabilities of web browser engines (e.g., embedded WebView component) would typically use the provided cache/persistence APIs in order to ensure that origin-bound data is preserved. This specification does not mandate particular mitigation strategies for dealing with size limitations (e.g., cache invalidation, storage reset), but merely recommends to follow best practices as used on the Open Web Platform.
+						</p>
+					</li>
+
+					<li>
+						<p>
+							Reading Systems that support scripting and network access should also include
+							methods to notify the user that network activity is occurring and/or that allow them to disable it.
+						</p>
+					</li>
+
+					<li>
+						<p>
+							If a Reading System allows users/scripts to store persistent data, it must treat that data as
 							sensitive. Scripts may save persistent data through cookies and DOM storage but Reading
 							Systems may block such attempts. Reading Systems that allow users to store data must ensure
 							they do not make that data available to other unrelated documents (e.g., ones that could be
 							spoofed). In particular, checking for a matching document identifier (or similar metadata)
-							is not a valid method to control access to persistent data.</p>
-						<p>Reading Systems that allow local storage should also provide methods for users to inspect,
+							is not a valid method to control access to persistent data.
+						</p>
+					</li>
+
+					<li>
+						<p>
+							Reading Systems that allow local storage should also provide methods for users to inspect,
 							disable, and delete that data. The Reading System must destroy the data if the corresponding
-							EPUB Publication is deleted.</p>
+							EPUB Publication is deleted from its "bookshelf".
```

Comment:
What if the reading system doesn't use a "bookshelf"? Shouldn't the requirement be the same if the user closes off such an application?

Reply:
Yep, true.

Maybe say:

> The Reading System must destroy the data if the corresponding EPUB Publication is deleted from its "bookshelf" if it implements one; otherwise it must destroy the data before the Reading System closes.

WDYT?

Reply:
SGTM

Reply:
Done in https://github.com/w3c/epub-specs/pull/1671/commits/a3cee4e2ae3e9b54e53d4866d7863604667cae05